org-code-javabuilder/lib/src/main/java/org/code/javabuilder/MainRunner.java

```diff
@@ -46,6 +46,18 @@ public void run(URLClassLoader urlClassLoader) throws JavabuilderException {
       if (e.getCause() instanceof FileNotFoundException) {
         throw new UserInitiatedException(UserInitiatedExceptionKey.FILE_NOT_FOUND, e.getCause());
       }
+      // NoClassDefFoundError is thrown by the class loader if the user attempts to use a disallowed
+      // class.
+      if (e.getCause() instanceof NoClassDefFoundError) {
```

Comment:
Should we handle this for tests too?

Reply:
This is trickier than I thought, starting a thread [here](https://codedotorg.slack.com/archives/C01EF4GJ9GE/p1637281178354400). May warrant a separate PR.